src/Storages/VirtualColumnUtils.cpp

```diff
@@ -115,7 +119,19 @@ NameSet getVirtualNamesForFileLikeStorage()
     return {"_path", "_file", "_size", "_time"};
 }
 
-VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription & storage_columns)
+std::map<std::string, std::string> parseHivePartitioningKeysAndValues(const std::string& path)
+{
+    std::string pattern = "/([^/]+)=([^/]+)";
+    re2::StringPiece input_piece(path);
+
+    std::map<std::string, std::string> key_values;
+    std::string key, value;
+    while (RE2::FindAndConsume(&input_piece, pattern, &key, &value))
+        key_values["_" + key] = value;
```

Comment:
We forgot to check if the table structure contains column `_key`. We should prepend another underscore to the name, until there is no ambiguity.

Also, probably it will be good to throw an exception if the path contains duplicated keys